specification/app/resource-manager/Microsoft.App/preview/2024-10-02-preview/examples/HttpRouteConfig_CreateOrUpdate.json

```diff
@@ -0,0 +1,130 @@
+{
+  "parameters": {
+    "subscriptionId": "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345",
+    "resourceGroupName": "examplerg",
+    "environmentName": "testcontainerenv",
+    "httpRouteName": "httproutefriendlyname",
+    "api-version": "2024-10-02-preview",
+    "httpRouteConfigEnvelope": {
+      "location": "East US",
+      "properties": {
+        "customDomains": [
+          {
+            "name": "example.com",
+            "bindingType": "SniEnabled",
+            "certificateId": "/subscriptions/34adfa4f-cedf-4dc0-ba29-b6d1a69ab345/resourceGroups/examplerg/providers/Microsoft.App/managedEnvironments/testcontainerenv/certificates/certificate-1"
+          }
+        ],
+        "rules": [
+          {
+            "targets": [
+              {
+                "containerAppName": "capp-1",
+                "revisionName": "rev-1",
+                "weight": 100
+              }
+            ],
+            "routes": [
+              {
+                "match": {
+                  "path": "/v1",
+                  "caseSensitive": true
+                },
+                "action": {
+                  "prefixRewrite": "/v1/api"
+                }
+              }
+            ],
+            "description": "random-description"
+          }
+        ]
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "headers": {},
+      "body": {
+        "id": "/subscriptions/34adfa4f-cedf-4dc0-ba29-b6d1a69ab345/resourceGroups/examplerg/providers/Microsoft.App/managedEnvironments/testcontainerenv/httpRouteConfigs/route-1",
+        "type": "Microsoft.App/managedEnvironments/httpRouteConfigs",
+        "location": "East US",
+        "properties": {
+          "fqdn": "app1.example.com",
+          "provisioningState": "InProgress",
+          "customDomains": [
+            {
+              "name": "example.com",
+              "bindingType": "SniEnabled",
+              "certificateId": "/subscriptions/34adfa4f-cedf-4dc0-ba29-b6d1a69ab345/resourceGroups/examplerg/providers/Microsoft.App/managedEnvironments/testcontainerenv/certificates/certificate-1"
+            }
+          ],
+          "rules": [
+            {
+              "targets": [
+                {
+                  "containerAppName": "capp-1",
+                  "revisionName": "rev-1",
+                  "weight": 100
+                }
+              ],
+              "routes": [
+                {
+                  "match": {
+                    "path": "/v1",
+                    "caseSensitive": true
+                  },
+                  "action": {
+                    "prefixRewrite": "/v1/api"
+                  }
+                }
+              ],
+              "description": "random-description"
+            }
+          ]
+        }
+      }
+    },
+    "201": {
```

Comment:
if it is not long running operation, you should never return 201. Please remove it

Reply:
i removed 201 and now it is failing lint validation

Reply:
![image](https://github.com/user-attachments/assets/8b7e6929-e5cc-4fba-bece-69dbaad1a1b9)

Reply:
We still need to adhere to synchronous running operation based on the source code of the lintDiff validation check
Hence, we need to add a 201 for non long runnning opps as well

https://github.com/Azure/azure-openapi-validator/blob/c26c69a9a4bc2c36a98752bb7dd89a7292e4a39a/packages/rulesets/src/spectral/functions/put-response-codes.ts#L39
![image](https://github.com/user-attachments/assets/4e4c8534-6d23-43ae-9093-8c32e114b7da)

Reply:
picking a random spec from another resource provider, they have one of their `PUT` that has `201` responses. 
https://github.com/tdaroly/azure-rest-api-specs/blob/tdarolywala/advanced-ingress-routing-2024-10-02-preview/specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2024-06-01-preview/vi.json
![image](https://github.com/user-attachments/assets/82301dfa-9654-4c6e-a710-e0e89964d889)

Reply:
The lint error is for patch and the example file you update "HttpConfig_CreateOrUpdate" is for put. They are not related. Please remove 201 for HttpConfig_CreateOrUpdate. If there is lint error occured, please also fix as mentioned.

Reply:
@Juliehzl i have fixed the lint error for PATCH. 

I am now referring to the other error pertaining to `201` status codes for non long running operations. It seems that ARM has added a validation that required 201 for those as well. Do we not want to support that? if so, yes i agree with you that we should ask ARM for exception.

Reply:
I see. If it is treated as valid for arm, then I am fine with it. Thanks!